ssc/cmod_wind_landbosse.cpp

```diff
@@ -0,0 +1,350 @@
+                                           "Use 'set_python_path' function in sscapi.h to point to the folder containing the file.");
+
+    landbosse_config_doc >> landbosse_config_root;
+
+    if (!landbosse_config_root.isMember("run_cmd"))
+        throw exec_error("wind_landbosse", "Missing key 'run_cmd' in 'landbosse.json'.");
+    if (!landbosse_config_root.isMember("min_python_version"))
+        throw exec_error("wind_landbosse", "Missing key 'min_python_version' in 'landbosse.json'.");
+
+    python_run_cmd = landbosse_config_root["run_cmd"].asString();
+    auto min_python_version = landbosse_config_root["min_python_version"].asString();
+
+    // check version works out
+    std::stringstream min_ver(min_python_version);
+    std::stringstream py_ver(python_version);
+    std::string min_ver_token, py_ver_token;
+
+    while (std::getline(min_ver, min_ver_token, '.')) {
+        if (!std::getline(py_ver, py_ver_token, '.'))
+            return;
+        if (std::stoi(min_ver_token) > std::stoi(py_ver_token))
+            throw exec_error("wind_landbosse", "'min_python_version' requirement not met.");
+    }
+}
+
+const size_t BUFSIZE = 4096;
+
+std::string cm_wind_landbosse::call_python_module(const std::string& input_dict_as_text){
+    std::promise<std::string> python_result;
+    std::future<std::string> f_completes = python_result.get_future();
+    std::thread([&]
+                {
+                    std::string cmd = std::string(get_python_path()) + "/" + python_exec_path + " -c \"" + python_run_cmd + "\"";
+                    size_t pos = cmd.find("<input>");
+                    cmd.replace(pos, 7, input_dict_as_text);
+
+                    FILE *file_pipe = popen(cmd.c_str(), "r");
+                    if (!file_pipe){
+                        python_result.set_value("wind_landbosse error. Could not call python with cmd:\n" + cmd);
+                        return;
+                    }
+
+                    std::string mod_response;
+                    char buffer[BUFSIZE];
+                    while (fgets(buffer, sizeof(buffer), file_pipe)){
+                        mod_response += buffer;
+                    }
+                    pclose(file_pipe);
+                    if (mod_response.empty())
+                        python_result.set_value("LandBOSSE error. Function did not return a response.");
+                    else
+                        python_result.set_value(mod_response);
+                }
+    ).detach();
+
+    std::chrono::system_clock::time_point time_passed
+            = std::chrono::system_clock::now() + std::chrono::seconds(60 * 5);
+
+    if(std::future_status::ready == f_completes.wait_until(time_passed))
+        return f_completes.get();
+    else
+        throw exec_error("wind_landbosse", "python handler error. Python process timed out.");
+}
+
+#ifdef __WINDOWS__
+std::string cm_wind_landbosse::call_python_module_windows(const std::string& input_dict_as_text) {
```

Comment:
Need to make this function use consistent alignment and curly braces.

Reply:
edited